src/lib_c/x86_64-linux-gnu/c/unistd.cr

```diff
@@ -43,4 +43,5 @@ lib LibC
   fun sysconf(name : Int) : Long
   fun unlink(name : Char*) : Int
   fun write(fd : Int, buf : Void*, n : SizeT) : SSizeT
+  fun sendfile(fd_out : Int, fd_in : Int, offset : OffT*, count : SizeT) : SSizeT
```

Comment:
Obviously more bindings need to be added and platforms which dont support it handled.

Reply:
of course, was just a start. BSD/Darwin support sendfile too but with two extra arguments (prepend/append data). not sure about windows but i assume there should something similar..

Reply:
Based on https://tinyclouds.org/iocp-links.html
* [Windows](https://docs.microsoft.com/en-us/windows/win32/api/mswsock/nf-mswsock-transmitfile?redirectedfrom=MSDN)
* [Linux](http://man7.org/linux/man-pages/man2/sendfile.2.html)
* [FreeBSD](https://www.freebsd.org/cgi/man.cgi?query=sendfile&sektion=2)
* [Darwin](http://www.manpagez.com/man/2/sendfile/)

How about adding a `Crystal::System.sendfile?(src : IO, dst : IO, limit : Int) : UInt64?` that returns `nil` if there was not a way to send the data from one `IO` to another in a specialized way; otherwise the number of bytes sent. 

The type checks would be done inside that definition and using it could be used like

```cr
len = Crystal::System.sendfile?(src, dst, limit)
return len if len
```

Reply:
*FreeBSD* supports sendfile. Not BSD in general.